cesium_app/util.py

```diff
@@ -47,49 +47,55 @@ def check_model_param_types(model_type, model_params, all_as_lists=False):
         Raises ValueError if parameter(s) are not of expected type.
 
     """
-    # Find relevant model description
-    for entry in model_descriptions:
-        if entry["name"] == model_type:
-            params_list = entry["params"]
-            break
-    else:
+    try:
+        model_desc = next(md for md in model_descriptions
+                              if md['name'] == model_type)
+    except StopIteration:
         raise ValueError("model_type not in list of allowable models.")
 
-    # Iterate through params and check against expected types
-    for k, v in model_params.items():
-        # Empty string or "None" goes to `None`
-        if v in ["None", ""]:
-            model_params[k] = None
-            continue
-        # Find relevant parameter description
-        for p in params_list:
-            if p["name"] == k:
-                param_entry = p
-                break
-        else:
-            raise ValueError("Parameter name not found in model description.")
-
-        # Combine logic for params with one type and multiple types
-        dest_types_list = make_list(param_entry["type"])
+
+    def verify_type(value, vtype):
+        """Ensure that value is of vtype.
+
+        Parameters
+        ----------
+        value : any
+            Input value.
+        vtype : type or list of types
+            Check that value is one of these types.
+
+        """
+        values = value if isinstance(value, list) else [value]
+
+        none_vals = [None, "None", ""]
+        values = [None if v in none_vals else v for v in values]
+
+        vtypes = set(make_list(vtype))
```

Comment:
Don't think the performance is important since it's so small; `x in some_list` is just as natural to me as `x in some_set` so I have no preference.